src/type.ts

```diff
@@ -335,15 +335,15 @@ export class Type {
 	}
 
 	get properties(): Property[] {
-		let propertiesArray: Property[] = [];
-		for (var type: Type = this; type != null; type = type.baseType) {
+		let propertiesObject: { [name: string]: Property } = this.__properties__;
+		for (var type: Type = this.baseType; type != null; type = type.baseType) {
 			for (var propertyName in type.__properties__) {
-				if (type.__properties__.hasOwnProperty(propertyName)) {
-					propertiesArray.push(type.__properties__[propertyName]);
+				if (!propertiesObject.hasOwnProperty(propertyName)) {
+					propertiesObject[propertyName] = type.__properties__[propertyName];
```

Comment:
We shouldn't be modifying the `__properties__` field.

Reply:
Updated, using spread to get `this.__properties__` in `propertiesObject`